diff --git a/saleor/core/utils/url.py b/saleor/core/utils/url.py
index f9fb6c374..16539b4ff 100644
--- a/saleor/core/utils/url.py
+++ b/saleor/core/utils/url.py
@@ -2,7 +2,8 @@ from urllib.parse import urlparse, urlsplit
 
 from django.conf import settings
 from django.core.exceptions import ValidationError
-from django.http.request import split_domain_port, validate_host
+from django.http.request import split_domain_port
+from tenants.utils import validate_client_host
 
 
 def validate_storefront_url(url):
@@ -20,7 +21,7 @@ def validate_storefront_url(url):
             )
     except ValueError as error:
         raise ValidationError(error)
-    if not validate_host(domain, settings.ALLOWED_CLIENT_HOSTS):
+    if not validate_client_host(domain):
         error_message = (
             f"{domain or url} is not allowed. Please check "
             "`ALLOWED_CLIENT_HOSTS` configuration."
diff --git a/saleor/graphql/account/tests/test_account.py b/saleor/graphql/account/tests/test_account.py
index 8a43753bd..5dca66eae 100644
--- a/saleor/graphql/account/tests/test_account.py
+++ b/saleor/graphql/account/tests/test_account.py
@@ -706,7 +706,7 @@ ACCOUNT_REGISTER_MUTATION = """
 
 
 @override_settings(
-    ENABLE_ACCOUNT_CONFIRMATION_BY_EMAIL=True, ALLOWED_CLIENT_HOSTS=["localhost"]
+    ENABLE_ACCOUNT_CONFIRMATION_BY_EMAIL=True
 )
 @patch("saleor.account.emails._send_account_confirmation_email")
 def test_customer_register(send_account_confirmation_email_mock, api_client):
@@ -714,7 +714,7 @@ def test_customer_register(send_account_confirmation_email_mock, api_client):
     variables = {
         "email": email,
         "password": "Password",
-        "redirectUrl": "http://localhost:3000",
+        "redirectUrl": "http://example.com",
     }
     query = ACCOUNT_REGISTER_MUTATION
     mutation_name = "accountRegister"
