diff --git a/saleor/core/jwt.py b/saleor/core/jwt.py
index aeed424ed..722bf5585 100644
--- a/saleor/core/jwt.py
+++ b/saleor/core/jwt.py
@@ -5,6 +5,7 @@ import graphene
 import jwt
 from django.conf import settings
 from django.core.handlers.wsgi import WSGIRequest
+from django.db import connection
 
 from ..account.models import User
 from ..app.models import App
@@ -21,9 +22,17 @@ JWT_REFRESH_TOKEN_COOKIE_NAME = "refreshToken"
 PERMISSIONS_FIELD = "permissions"
 
 
+def jwt_issuer():
+    return connection.tenant.domain_url
+
+
+def jwt_secret():
+    return connection.tenant.jwt_secret_key
+
+
 def jwt_base_payload(exp_delta: timedelta) -> Dict[str, Any]:
     utc_now = datetime.utcnow()
-    payload = {"iat": utc_now, "exp": utc_now + exp_delta}
+    payload = {"iat": utc_now, "exp": utc_now + exp_delta, "iss": jwt_issuer()}
     return payload
 
 
@@ -52,7 +61,7 @@ def jwt_user_payload(
 def jwt_encode(payload: Dict[str, Any]) -> str:
     return jwt.encode(
         payload,
-        settings.SECRET_KEY,  # type: ignore
+        jwt_secret(),
         JWT_ALGORITHM,
     ).decode("utf-8")
 
@@ -60,9 +69,10 @@ def jwt_encode(payload: Dict[str, Any]) -> str:
 def jwt_decode(token: str) -> Dict[str, Any]:
     return jwt.decode(
         token,
-        settings.SECRET_KEY,  # type: ignore
+        jwt_secret(),
         algorithms=JWT_ALGORITHM,
         verify_expiration=settings.JWT_EXPIRE,
+        issuer=jwt_issuer(),
     )
 
 
