diff --git a/saleor/csv/tests/export/products_data/test_get_products_data.py b/saleor/csv/tests/export/products_data/test_get_products_data.py
index 0d9468d90..fd1977cc0 100644
--- a/saleor/csv/tests/export/products_data/test_get_products_data.py
+++ b/saleor/csv/tests/export/products_data/test_get_products_data.py
@@ -113,6 +113,12 @@ def test_get_products_data(product, product_with_image, collection, image, chann
             data = add_channel_to_expected_variant_data(data, variant, channel_ids)
 
             expected_data.append(data)
+    expected_data[0]["media__image"] = expected_data[0]["media__image"].replace(
+        "http://mirumee.com/media/mirumee.com/", "http://mirumee.com/media/"
+    )
+    expected_data[0]["variants__media__image"] = expected_data[0][
+        "variants__media__image"
+    ].replace("http://mirumee.com/media/mirumee.com/", "http://mirumee.com/media/")
     assert result_data == expected_data
 
 
diff --git a/saleor/graphql/account/tests/test_account.py b/saleor/graphql/account/tests/test_account.py
index dda748fd9..ed2eed114 100644
--- a/saleor/graphql/account/tests/test_account.py
+++ b/saleor/graphql/account/tests/test_account.py
@@ -3716,7 +3716,7 @@ def test_user_avatar_update_mutation(monkeypatch, staff_api_client, media_root):
 
     assert user.avatar
     assert data["user"]["avatar"]["url"].startswith(
-        "http://testserver/media/user-avatars/avatar"
+        "http://mirumee.com/media/mirumee.com/user-avatars/avatar"
     )
     img_name, format = os.path.splitext(image_file._name)
     file_name = user.avatar.name
@@ -3743,7 +3743,7 @@ def test_user_avatar_update_mutation_image_exists(staff_api_client, media_root):
 
     assert user.avatar != avatar_mock
     assert data["user"]["avatar"]["url"].startswith(
-        "http://testserver/media/user-avatars/new_image"
+        "http://mirumee.com/media/mirumee.com/user-avatars/new_image"
     )


diff --git a/saleor/graphql/core/tests/test_file_upload.py b/saleor/graphql/core/tests/test_file_upload.py
index cf29e0563..2e5c3f973 100644
--- a/saleor/graphql/core/tests/test_file_upload.py
+++ b/saleor/graphql/core/tests/test_file_upload.py
@@ -149,6 +149,7 @@ def test_file_upload_file_with_the_same_name_already_exists(
     assert not errors
     assert data["uploadedFile"]["contentType"] == "image/png"
     file_url = data["uploadedFile"]["url"]
-    assert file_url != "http://testserver/media/" + image_file._name
-    assert file_url != "http://testserver/media/" + path
-    assert default_storage.exists(file_url.replace("http://testserver/media/", ""))
+    assert file_url != "http://mirumee.com/media/" + image_file._name
+    assert file_url != "http://mirumee.com/media/" + path
+    assert file_url.startswith("http://mirumee.com/media/")
+    assert default_storage.exists(file_url.replace("http://mirumee.com/media/", ""))


diff --git a/saleor/graphql/page/tests/test_page.py b/saleor/graphql/page/tests/test_page.py
index 3f54c2c2d..fde83d207 100644
--- a/saleor/graphql/page/tests/test_page.py
+++ b/saleor/graphql/page/tests/test_page.py
@@ -496,7 +496,7 @@ def test_create_page_with_file_attribute(
                 "slug": f"{attr_value.slug}-2",
                 "name": attr_value.name,
                 "file": {
-                    "url": f"http://testserver/media/{attr_value.file_url}",
+                    "url": f"http://mirumee.com/media/{attr_value.file_url}",
                     "contentType": None,
                 },
                 "reference": None,
@@ -570,7 +570,7 @@ def test_create_page_with_file_attribute_new_attribute_value(
                 "reference": None,
                 "name": new_value,
                 "file": {
-                    "url": "http://testserver/media/" + new_value,
+                    "url": "http://mirumee.com/media/" + new_value,
                     "contentType": new_value_content_type,
                 },
             }
@@ -1285,7 +1285,7 @@ def test_update_page_with_file_attribute_value(
                 "name": new_value,
                 "reference": None,
                 "file": {
-                    "url": "http://testserver/media/" + new_value,
+                    "url": "http://mirumee.com/media/" + new_value,
                     "contentType": None,
                 },
             }
@@ -1338,7 +1338,7 @@ def test_update_page_with_file_attribute_new_value_is_not_created(
                 "name": existing_value.name,
                 "reference": None,
                 "file": {
-                    "url": f"http://testserver/media/{existing_value.file_url}",
+                    "url": f"http://mirumee.com/media/{existing_value.file_url}",
                     "contentType": existing_value.content_type,
                 },
             }


diff --git a/saleor/graphql/product/tests/test_product.py b/saleor/graphql/product/tests/test_product.py
index 5241f9c..06e09b0 100644
--- a/saleor/graphql/product/tests/test_product.py
+++ b/saleor/graphql/product/tests/test_product.py
@@ -3616,7 +3616,7 @@ def test_create_product_with_file_attribute(
                     "name": existing_value.name,
                     "slug": f"{existing_value.slug}-2",
                     "file": {
-                        "url": f"http://testserver/media/{existing_value.file_url}",
+                        "url": f"http://mirumee.com/media/{existing_value.file_url}",
                         "contentType": None,
                     },
                     "reference": None,
@@ -3905,7 +3905,7 @@ def test_create_product_with_file_attribute_new_attribute_value(
                     "richText": None,
                     "boolean": None,
                     "file": {
-                        "url": "http://testserver/media/" + non_existing_value,
+                        "url": "http://mirumee.com/media/" + non_existing_value,
                         "contentType": None,
                     },
                 }
@@ -5214,7 +5214,7 @@ def test_update_product_with_file_attribute_value(
                 "slug": slugify(new_value),
                 "reference": None,
                 "file": {
-                    "url": "http://testserver/media/" + new_value,
+                    "url": "http://mirumee.com/media/" + new_value,
                     "contentType": None,
                 },
                 "boolean": None,
@@ -5276,7 +5276,7 @@ def test_update_product_with_file_attribute_value_new_value_is_not_created(
                 "slug": existing_value.slug,
                 "reference": None,
                 "file": {
-                    "url": f"http://testserver/media/{existing_value.file_url}",
+                    "url": f"http://mirumee.com/media/{existing_value.file_url}",
                     "contentType": existing_value.content_type,
                 },
                 "boolean": None,


diff --git a/saleor/graphql/product/tests/test_variant.py b/saleor/graphql/product/tests/test_variant.py
index 028dbd600..1f0f6992f 100644
--- a/saleor/graphql/product/tests/test_variant.py
+++ b/saleor/graphql/product/tests/test_variant.py
@@ -1913,7 +1913,7 @@ def test_update_product_variant_with_file_attribute_new_value_is_not_created(
     assert value_data["name"] == existing_value.name
     assert (
         value_data["file"]["url"]
-        == f"http://testserver/media/{existing_value.file_url}"
+        == f"http://mirumee.com/media/{existing_value.file_url}"
     )
     assert value_data["file"]["contentType"] == existing_value.content_type

