diff --git a/conftest.py b/conftest.py
index 62f7fb73b..4e4fc1776 100644
--- a/conftest.py
+++ b/conftest.py
@@ -1,7 +1,54 @@
+import pytest
+from django.db import connection
+from tenant_schemas.utils import get_tenant_model
+from tenants.postgresql_backend.creation import DEFAULT_TEST_TENANT, OTHER_TEST_TENANT
+from tenants.tests.client import TenantClient
+
+
 pytest_plugins = [
+    "pytester",
+    "tenants.tests.fixtures",
     "saleor.tests.fixtures",
     "saleor.plugins.tests.fixtures",
     "saleor.graphql.tests.fixtures",
     "saleor.graphql.channel.tests.fixtures",
     "saleor.graphql.order.tests.benchmark.fixtures",
 ]
+
+
+def pytest_configure(config):
+    """Flags python as being running inside a test suite.
+
+    Pattern from https://is.gd/hXRqGq (pytest documentation).
+    Needed by `tenants.management.commands.migrate`.
+    """
+    import sys
+
+    sys._called_from_test = True
+
+
+def pytest_unconfigure(config):
+    """Un-flags python as being running inside a test suite."""
+    import sys
+
+    del sys._called_from_test
+
+
+@pytest.fixture
+def test_tenant(db):
+    return get_tenant_model().objects.get(**DEFAULT_TEST_TENANT)
+
+
+@pytest.fixture
+def other_tenant(db):
+    return get_tenant_model().objects.get(**OTHER_TEST_TENANT)
+
+
+@pytest.fixture
+def client(test_tenant):
+    yield TenantClient(test_tenant)
+
+
+@pytest.fixture
+def other_client(other_tenant):
+    yield TenantClient(other_tenant)
diff --git a/saleor/graphql/tests/fixtures.py b/saleor/graphql/tests/fixtures.py
index 422233910b..8ace0dd446 100644
--- a/saleor/graphql/tests/fixtures.py
+++ b/saleor/graphql/tests/fixtures.py
@@ -5,6 +5,7 @@ import graphene
 import pytest
 from django.contrib.auth.models import AnonymousUser
 from django.core.serializers.json import DjangoJSONEncoder
+from django.db import connection
 from django.shortcuts import reverse
 from django.test.client import MULTIPART_CONTENT, Client

@@ -13,6 +14,7 @@ from ...core.jwt import create_access_token
 from ...tests.utils import flush_post_commit_hooks
 from ..views import handled_errors_logger, unhandled_errors_logger
 from .utils import assert_no_permission
+from tenants.tests.client import TenantClient

 API_PATH = reverse("api")
 ACCESS_CONTROL_ALLOW_ORIGIN = "Access-Control-Allow-Origin"
@@ -21,10 +23,11 @@ ACCESS_CONTROL_ALLOW_HEADERS = "Access-Control-Allow-Headers"
 ACCESS_CONTROL_ALLOW_METHODS = "Access-Control-Allow-Methods"


-class ApiClient(Client):
+class TenantApiClient(TenantClient):
     """GraphQL API client."""

     def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
         user = kwargs.pop("user", AnonymousUser())
         app = kwargs.pop("app", None)
         self._user = None
@@ -32,12 +35,9 @@ class ApiClient(Client):
         self.user = user
         self.app_token = None
         self.app = app
-        if not user.is_anonymous:
-            self.token = create_access_token(user)
-        elif app:
+        if app:
             token = app.tokens.first()
             self.app_token = token.auth_token if token else None
-        super().__init__(*args, **kwargs)

     def _base_environ(self, **request):
         environ = super()._base_environ(**request)
@@ -118,27 +118,32 @@ class ApiClient(Client):

 @pytest.fixture
 def app_api_client(app):
-    return ApiClient(app=app)
+    return TenantApiClient(connection.tenant, app=app)


 @pytest.fixture
 def staff_api_client(staff_user):
-    return ApiClient(user=staff_user)
+    return TenantApiClient(connection.tenant, user=staff_user)


 @pytest.fixture
 def superuser_api_client(superuser):
-    return ApiClient(user=superuser)
+    return TenantApiClient(connection.tenant, user=superuser)


 @pytest.fixture
 def user_api_client(customer_user):
-    return ApiClient(user=customer_user)
+    return TenantApiClient(connection.tenant, user=customer_user)


 @pytest.fixture
 def api_client():
-    return ApiClient(user=AnonymousUser())
+    return TenantApiClient(connection.tenant, user=AnonymousUser())
+
+
+@pytest.fixture
+def other_tenant_api_client(other_tenant):
+    return TenantApiClient(other_tenant, user=AnonymousUser)


 @pytest.fixture
diff --git a/saleor/tests/fixtures.py b/saleor/tests/fixtures.py
index 571b4020d..61f68ec68 100644
--- a/saleor/tests/fixtures.py
+++ b/saleor/tests/fixtures.py
@@ -224,12 +224,13 @@ def site_settings(db, settings) -> SiteSettings:
     saleor.site.models.SiteSettings have a one-to-one relationship and a site
     should never exist without a matching settings object.
     """
-    site = Site.objects.get_or_create(name="mirumee.com", domain="mirumee.com")[0]
+    site = Site.objects.get_or_create(domain="mirumee.com")[0]
+    site.name = "mirumee.com"
+    site.save()
     obj = SiteSettings.objects.get_or_create(
-        site=site,
-        default_mail_sender_name="Mirumee Labs",
-        default_mail_sender_address="mirumee@example.com",
+        site=site
     )[0]
+
     settings.SITE_ID = site.pk

     main_menu = Menu.objects.get_or_create(
@@ -242,6 +243,8 @@ def site_settings(db, settings) -> SiteSettings:
     )[0]
     obj.top_menu = main_menu
     obj.bottom_menu = secondary_menu
+    obj.default_mail_sender_name = "Mirumee Labs"
+    obj.default_mail_sender_address = "mirumee@example.com"
     obj.save()
     return obj

