diff --git a/saleor/csv/tests/export/products_data/test_get_products_data.py b/saleor/csv/tests/export/products_data/test_get_products_data.py
index 5c1bcea68..ba273e1c2 100644
--- a/saleor/csv/tests/export/products_data/test_get_products_data.py
+++ b/saleor/csv/tests/export/products_data/test_get_products_data.py
@@ -113,6 +113,12 @@ def test_get_products_data(product, product_with_image, collection, image, chann
             data = add_channel_to_expected_variant_data(data, variant, channel_ids)

             expected_data.append(data)
+    expected_data[0]["images__image"] = expected_data[0]["images__image"].replace(
+        "http://mirumee.com/media/mirumee.com/", "http://mirumee.com/media/"
+    )
+    expected_data[0]["variants__images__image"] = expected_data[0][
+        "variants__images__image"
+    ].replace("http://mirumee.com/media/mirumee.com/", "http://mirumee.com/media/")
     assert result_data == expected_data


diff --git a/saleor/csv/tests/test_emails.py b/saleor/csv/tests/test_emails.py
index a118696c1..a0d5cc7f8 100644
--- a/saleor/csv/tests/test_emails.py
+++ b/saleor/csv/tests/test_emails.py
@@ -26,7 +26,7 @@ def test_send_email_with_link_to_download_file(
     # then
     template = emails.EXPORT_TEMPLATES["export_products_success"]
     ctx = {
-        "csv_link": f"http://mirumee.com/media/export_files/{file_mock.name}",
+        "csv_link": f"http://mirumee.com/media/mirumee.com/export_files/{file_mock.name}",
         "domain": "mirumee.com",
         "site_name": "mirumee.com",
     }
diff --git a/saleor/graphql/account/tests/test_account.py b/saleor/graphql/account/tests/test_account.py
index eb9574ae6..96eee7512 100644
--- a/saleor/graphql/account/tests/test_account.py
+++ b/saleor/graphql/account/tests/test_account.py
@@ -3491,7 +3491,7 @@ def test_user_avatar_update_mutation(monkeypatch, staff_api_client, media_root):

     assert user.avatar
     assert data["user"]["avatar"]["url"].startswith(
-        "http://testserver/media/user-avatars/avatar"
+        "http://mirumee.com/media/mirumee.com/user-avatars/avatar"
     )

     # The image creation should have triggered a warm-up
@@ -3518,7 +3518,7 @@ def test_user_avatar_update_mutation_image_exists(staff_api_client, media_root):

     assert user.avatar != avatar_mock
     assert data["user"]["avatar"]["url"].startswith(
-        "http://testserver/media/user-avatars/new_image"
+        "http://mirumee.com/media/mirumee.com/user-avatars/new_image"
     )


diff --git a/saleor/graphql/core/tests/test_file_upload.py b/saleor/graphql/core/tests/test_file_upload.py
index 9d8b8cef2..4ea6747ed 100644
--- a/saleor/graphql/core/tests/test_file_upload.py
+++ b/saleor/graphql/core/tests/test_file_upload.py
@@ -38,7 +38,7 @@ def test_file_upload_by_staff(staff_api_client, site_settings, media_root):
     data = content["data"]["fileUpload"]
     errors = data["uploadErrors"]

-    expected_path = "http://testserver/media/" + image_file._name
+    expected_path = "http://mirumee.com/media/" + image_file._name
     assert not errors
     assert data["uploadedFile"]["contentType"] == "image/png"
     assert data["uploadedFile"]["url"] == expected_path
@@ -76,7 +76,7 @@ def test_file_upload_by_app(app_api_client, media_root):
     data = content["data"]["fileUpload"]
     errors = data["uploadErrors"]

-    expected_path = "http://testserver/media/" + image_file._name
+    expected_path = "http://mirumee.com/media/" + image_file._name
     assert not errors
     assert data["uploadedFile"]["contentType"] == "image/png"
     assert data["uploadedFile"]["url"] == expected_path
@@ -99,7 +99,7 @@ def test_file_upload_by_superuser(superuser_api_client, media_root):
     data = content["data"]["fileUpload"]
     errors = data["uploadErrors"]

-    expected_path = "http://testserver/media/" + image_file._name
+    expected_path = "http://mirumee.com/media/" + image_file._name
     assert not errors
     assert data["uploadedFile"]["contentType"] == "image/png"
     assert data["uploadedFile"]["url"] == expected_path
@@ -137,6 +137,6 @@ def test_file_upload_file_with_the_same_name_already_exists(
     assert not errors
     assert data["uploadedFile"]["contentType"] == "image/png"
     file_url = data["uploadedFile"]["url"]
-    assert file_url != "http://testserver/media/" + image_file._name
-    assert file_url != "http://testserver/media/" + path
-    assert default_storage.exists(file_url.replace("http://testserver/media/", ""))
+    assert file_url != "http://mirumee.com/media/" + image_file._name
+    assert file_url != "http://mirumee.com/media/" + path
+    assert default_storage.exists(file_url.replace("http://mirumee.com/media/", ""))
diff --git a/saleor/graphql/page/tests/test_page.py b/saleor/graphql/page/tests/test_page.py
index c55de6e69..c06e3a041 100644
--- a/saleor/graphql/page/tests/test_page.py
+++ b/saleor/graphql/page/tests/test_page.py
@@ -481,8 +481,8 @@ def test_create_page_with_file_attribute_new_attribute_value(
                 "slug": slugify(new_value),
                 "reference": None,
                 "name": new_value,
                 "file": {
-                    "url": "http://testserver/media/" + new_value,
+                    "url": "http://mirumee.com/media/" + new_value,
                     "contentType": new_value_content_type,
                 },
             }
@@ -751,8 +751,8 @@ def test_update_page_with_file_attribute_value(
                 "slug": slugify(new_value),
                 "name": new_value,
                 "reference": None,
                 "file": {
-                    "url": "http://testserver/media/" + new_value,
+                    "url": "http://mirumee.com/media/" + new_value,
                     "contentType": None,
                 },
             }
diff --git a/saleor/graphql/product/tests/test_product.py b/saleor/graphql/product/tests/test_product.py
index 2faf1c6dc..565a296ef 100644
--- a/saleor/graphql/product/tests/test_product.py
+++ b/saleor/graphql/product/tests/test_product.py
@@ -2374,8 +2374,8 @@ def test_create_product_with_file_attribute_new_attribute_value(
                     "name": non_existing_value,
                     "slug": slugify(non_existing_value, allow_unicode=True),
                     "reference": None,
                     "file": {
-                        "url": "http://testserver/media/" + non_existing_value,
+                        "url": "http://mirumee.com/media/" + non_existing_value,
                         "contentType": None,
                     },
                 }
@@ -3279,8 +3279,8 @@ def test_update_product_with_file_attribute_value(
                 "name": new_value,
                 "slug": slugify(new_value),
                 "reference": None,
                 "file": {
-                    "url": "http://testserver/media/" + new_value,
+                    "url": "http://mirumee.com/media/" + new_value,
                     "contentType": None,
                 },
             }
