diff --git a/saleor/graphql/views.py b/saleor/graphql/views.py
index f0a1755b2..6b6263c80 100644
--- a/saleor/graphql/views.py
+++ b/saleor/graphql/views.py
@@ -207,7 +207,7 @@ class GraphQLView(View):
     def execute_graphql_request(self, request: HttpRequest, data: dict):
         with opentracing.global_tracer().start_active_span("graphql_query") as scope:
             span = scope.span
-            span.set_tag(opentracing.tags.COMPONENT, "GraphQL")
+            span.set_tag(opentracing.tags.COMPONENT, "graphql")

             query, variables, operation_name = self.get_graphql_params(request, data)

